src/solvers/strings/format_specifier.cpp

```diff
@@ -0,0 +1,65 @@
+/*******************************************************************\
+
+Module: Format specifers for String.format
+
+Author: Romain Brenguier, Joel Allred
+
+Date:   June 2019
+
+\*******************************************************************/
+
+/// \file
+///  Format specifiers for String.format
+
+#include "format_specifier.h"
+#include <regex>
+
+/// Helper function for parsing format strings.
+/// This follows the implementation in openJDK of the java.util.Formatter class:
+/// http://hg.openjdk.java.net/jdk7/jdk7/jdk/file/9b8c96f96a0f/src/share/classes/java/util/Formatter.java#l2660
+/// \param m: a match in a regular expression
+/// \return Format specifier represented by the matched string. The groups in
+///   the match should represent: index, flag, width, precision, date and
+///   conversion type.
+static format_specifiert format_specifier_of_match(std::smatch &m)
+{
+  int index = m[1].str().empty() ? -1 : std::stoi(m[1].str());
+  std::string flag = m[2].str();
+  int width = m[3].str().empty() ? -1 : std::stoi(m[3].str());
+  int precision = m[4].str().empty() ? -1 : std::stoi(m[4].str());
```

Comment:
Is `int` really the most appropriate type? If anything, an `ssize_t` would seem better suited, but maybe even `optionalt<std::size_t>` to avoid the `-1` hack?